FirefoxAction.cs

```diff
@@ -114,14 +138,21 @@ internal static string GetStringValue(this RegistryKey registryKey, string name,
             }
         }
 
-        internal static bool ContainsName(this RegistryKey key, string name)
+        internal static JObject GetJSON(this RegistryKey registryKey, string name, string defaultValue = null)
```

Comment:
This is quite neat. It is not easy to follow, but the `new JObject()` return works subtly in both `ExtensionSettingsRemove` and `ExtensionSettingsInstall` cases, which is cool.